tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,699 @@
+package steps
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	resty "github.com/go-resty/resty/v2"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var namespace = "k8ssandra"
+var repairId string
+
+func RunShellCommand(command *exec.Cmd) error {
+	err := command.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = command.Wait()
+	return err
+}
+
+func RunShellCommandAndGetOutput(command *exec.Cmd) string {
+	var outb bytes.Buffer
+	command.Stdout = &outb
+	err := command.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(outb.String())
+}
+
+func GetServicesWithLabel(label string) (*v1.ServiceList, error) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	clientset, err := k8s.GetKubernetesClientFromOptionsE(GinkgoT(), kubectlOptions)
+	Expect(err).To(BeNil())
+	services, err := clientset.CoreV1().Services(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: label})
+	return services, err
+}
+
+// Find returns true if val exists in the slice array, false otherwise
+func Find(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
+
+func deployCluster(customValues string, helmValues map[string]string) {
+	clusterChartPath, err := filepath.Abs("../../charts/k8ssandra")
+	if err != nil {
+		Fail("Couldn't find the absolute path for K8ssandra charts")
+	}
+
+	customChartPath, err := filepath.Abs("charts/" + customValues)
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't find the absolute path for custom values: %s", customValues))
+	}
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	helmOptions := &helm.Options{}
+
+	helmOptions = &helm.Options{
+		// Enable traefik to allow redirections for testing
+		SetValues:      helmValues,
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+		ValuesFiles:    []string{customChartPath},
+	}
+
+	releaseName := "k8ssandra"
+	defer timeTrack(time.Now(), fmt.Sprintf("Installing and starting k8ssandra"))
+	helm.Install(GinkgoT(), helmOptions, clusterChartPath, releaseName)
+
+	// Wait for cass-operator pod to be ready
+	attempts := 0
+	maxAttempts := 10
```

Comment:
Why not ``for i := 0; i < maxAttempts; i++ { `` and exit if done?

Reply:
Also, if using gomega, you should use Eventually func. And ``--wait`` should not exit at all, so why multiple attempts?

Reply:
Actually, why on earth this part at all?

Reply:
> why on earth this part at all?

Please be welcoming and positive in your comments. No project wants contributors to feel offended.
If you want to teach me something, which is fine as I'm eager to learn, do it nicely. Thanks.

`Eventually` is a great alternative that I didn't know of indeed. It'll make my clunky code much cleaner and efficient.

Reply:
No, what I mean is that why do we have a part that repeats wait? Why not just set a longer timeout?

Reply:
This includes any unnecessary use of Eventually also.

Reply:
There are bits where multiple attempts are needed and we don't have a blocking call such as `kubectl wait`. 
In order to remove the magic sleep during restore, I'm waiting on cassandra operator progress to change to "Updating" and then "Ready" for example, which requires to repeat the test as long as we don't reach the expected value.
I'll push another commit soon with these changes and you can see if that matches your expectations.

Reply:
Actually not, because when restoring a backup, the cassandradatacenter already has the Ready condition but then gets updated by cass-operator.
Just checking for the Ready condition made the `wait` command to return instantly, without giving time for the Cassandra pods to be recreated and failed afterwards when trying to access Cassandra using cqlsh.
That's why I initially added the sleep...

So checking first for a state change (in this case detect that cass-operator is updating the cassandradatacenter) and then wait for the dc to be ready allows to remove the sleep and follow the chain of events properly.